apps/tx-builder/src/components/Builder.tsx

```diff
@@ -0,0 +1,319 @@
+    setReviewing(false);
+  };
+
+  const handleDismiss = () => {
+    setReviewing(false);
+  };
+
+  const addTransaction = async () => {
+    let description = '';
+    let data = '';
+    const web3 = services.web3;
+
+    if (!web3) {
+      return;
+    }
+
+    if (contract && contract.methods.length > selectedMethodIndex) {
+      const method = contract.methods[selectedMethodIndex];
+      const cleanInputs = [];
+      description = method.name + ' (';
+      for (let i = 0; i < method.inputs.length; i++) {
+        const cleanValue = inputCache[i] || '';
+        cleanInputs[i] = cleanValue.charAt(0) === '[' ? JSON.parse(cleanValue.replace(/"/g, '"')) : cleanValue;
+        if (i > 0) {
+          description += ', ';
+        }
+        const input = method.inputs[i];
+        description += (input.name || input.type) + ': ' + cleanValue;
+      }
+      description += ')';
+
+      try {
+        if (!['receive', 'fallback'].includes(method.name)) {
+          data = web3.eth.abi.encodeFunctionCall(method as AbiItem, cleanInputs);
+        }
+      } catch (error) {
+        setAddTxError(error.message);
+        return;
+      }
+    }
+
+    try {
+      const cleanTo = web3.utils.toChecksumAddress(toInput);
+      const cleanValue = valueInput.length > 0 ? web3.utils.toWei(valueInput) : 0;
+      if (data.length === 0) {
+        data = '0x';
+      }
+      if (description.length === 0) {
+        description = `Transfer ${web3.utils.fromWei(cleanValue.toString())} ETH to ${cleanTo}`;
+      }
+      transactions.push({
+        description,
+        raw: { to: cleanTo, value: cleanValue, data },
+      });
+      setInputCache([]);
+      setTransactions(transactions);
+      setSelectedMethodIndex(0);
+      setValueInput('');
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const deleteTransaction = async (inputIndex: number) => {
+    const newTxs = transactions.slice();
+    newTxs.splice(inputIndex, 1);
+    setTransactions(newTxs);
+  };
+
+  const isValidAddress = useCallback(
+    (address: string | null) => {
+      if (!address) {
+        return false;
+      }
+      return services?.web3?.utils.isAddress(address);
+    },
+    [services],
+  );
+
+  const sendTransactions = useCallback(async () => {
+    if (!transactions.length) {
+      return;
+    }
+
+    sdk.txs.send({ txs: transactions.map((d) => d.raw) }).catch(console.error);
+  }, [sdk, transactions]);
+
+  // Set toInput when to changes
+  useEffect(() => {
+    const value = isValidAddress(to) ? (to as string) : '';
+    setToInput(value);
+  }, [to, isValidAddress]);
+
+  // set when inputValue is visible
+  useEffect(() => {
+    const isVisible = async () => {
+      if (contract) {
+        const method = getContractMethod();
+        setIsValueInputVisible(method?.payable || false);
+      } else {
+        try {
+          const code = await services?.web3?.eth.getCode(toInput);
+          setIsValueInputVisible(code === '0x');
```

Comment:
Why are we doing this? Couldn't we just allow it when we don't have an abi? We cannot really say if it is possible or not to send eth ... not sure if defaulting to "not allowing" it for contracts is the best choice.